src/thunderbots/software/geom/util.h

```diff
@@ -486,3 +511,54 @@ Point getPointsMean(const std::vector<Point> &points);
  * @return the variance of the list of points
  */
 double getPointsVariance(const std::vector<Point> &points);
+
+
+/**
+ * Function returns the segment defined by the segment between the intersections of two
+ * Rays on a segment
+ *
+ * @param ray1 (Starting point and direction)
+ * @param ray2 (Starting point and direction)
+ * @param segment (Segment to find segment of intersection upon)
+ * @return Segment, the segment defined by the space between two intersecting rays on the
+ * segment parameter std::nullopt, if both rays don't intersect the segment, and the
+ * segment is not enclosed between the rays Segment, if one ray intersects the segment,
+ * but one of the segment parameters extremes are enclosed within the two rays
+ */
+std::optional<Segment> getIntersectingSegment(Ray ray1, Ray ray2, Segment segment);
+
+/**
+ * Function calculates whether the segment parameter is enclosed between the ray
```

Comment:
I don't entirely understand what it means for a segment to be "enclosed" between rays here?

Reply:
If you where to connect the theoretical ends of the rays together, the segment would be entirely within the area of that cone. I've added a picture to help describe this